### DISCUSSION TITLE
Welcome to ESP-Parking-Assistant Discussions!

### DISCUSSION THREAD

**Discussion by Resinchem** [Resinchem/ESP-Parking-Assistant#18] [Announcements]
## Welcome!
  Please use discussions for the following:
  * Ask questions you’re wondering about.
  * Troubleshoot your installation
  * Share ideas.
  * Engage with other community members.

**Comment by shomritab** [comment]
Hello and many thanks for the video and the article. I created this for my garage and it works perfect. Although there is some issue with it connecting to HomeAssistant via MQTT. I provide all the details of my MQTT broker, and saved, but I dont see anything in Home Assistant. Just curious if you could help me with this. Thanks.

**Reply by Resinchem** [reply]
Congrats on building the system!  As far as MQTT, you said that you set up the broker information in the parking app, but have you also defined the MQTT sensors in your Home Assistant configuration?  Currently, the parking app does not have Home Assistant discovery (it's something I'd like to add if I can ever find the time).  Until that time, it is necessary to define the sensors with YAML in your Home Assistant configuration.  You can find example code of how to do this in the [/examples](https://github.com/Resinchem/ESP-Parking-Assistant/tree/main/examples) folder. 

Make sure you are defining these under the mqtt: integration header.  If you don't have an mqtt: integration header in your configuration, then you will need to add it.

If you have added the YAML and defined the sensors, but still don't see them listed under the Developer Tools -> States in Home Assistant, let me know and we can troubleshoot a little further.

**Reply by shomritab** [reply]
Thank you for the quick reply. I was surely missing the sensor definitions. But even after I added them to my configurations.yaml as per the example, restarted both HA and park assistant, still I dont see the sensors under MQTT integrations. I even don't see anything on the MQTT explorer. Not sure what else is missing.

**Reply by Resinchem** [reply]
If you aren't seeing any topics/payloads published when using MQTT Explorer, then the parking assistant isn't connecting/sending data to your broker... and therefore no data will be seen in Home Assistant.

Double check your MQTT settings in the parking assistant web interface.  When you go to the main page of the web interface, do you see the MQTT settings you entered previously?  If the settings are reset, then they are not being saved.  Be sure to check the box to save your settings as the new boot defaults after entering in your MQTT details.  Also verify your MQTT settings are correct as stated in the [MQTT](https://github.com/Resinchem/ESP-Parking-Assistant/wiki/08-MQTT) topic of the wiki. 

Then check MQTT Explorer again... remembering that it may take a little time before you see any data published, based on the telemetry period you specified (60 seconds minimum... 600 seconds maximum).  Until you see the data being published under the topic you defined in MQTT Explorer, you will not be able to get the data/sensors into Home Assistant.  Let me know what you find.

**Reply by shomritab** [reply]
Its weird that I am doing exactly what you mentioned, but still dont see any topics on MQTT explorer. Do you suspect that it might be an issue with the D1 mini ? Although everything else, works just the same as you show in the youtube video. My bad that I didn't test MQTT before soldering everything. In any case its not a deal breaker for me. I am still happy with the system, and would use it without having a sensor in HA.

**Reply by Resinchem** [reply]
If the parking assistant is on your wifi (and I'm assuming it is since you can get to the web interface) and this is the same network as the MQTT broker, there is no reason that MQTT shouldn't be working.  I'm glad to try to help you troubleshoot if you like.

First, can you grab a screen shot of your MQTT settings in the parking app:
![image](https://github.com/Resinchem/ESP-Parking-Assistant/assets/55962781/23c8d1b9-4ba1-43eb-a4e1-70fc759e87f4)

Then can you also include the connection settings that you have setup in MQTT Explorer:  
![image](https://github.com/Resinchem/ESP-Parking-Assistant/assets/55962781/69fdb40b-aa7d-43d4-a259-f88850068dcd)

I'd really like to figure out why your MQTT isn't working.  I don't think it is an issue with the firmware (you are running the latest release, v0.44, right?), but it is always possible there is something unique in your setup that I haven't accounted for.  Along those lines, what are you using for your MQTT broker?  The Home Asssistant add-on, or do you have a standalone broker of some type?

**Reply by shomritab** [reply]
Sure, here are the details.
- Yes, I am running the latest firmware  v0.44
- Yes, I am using the - Home Assistant Add-on: Mosquitto broker

![image](https://github.com/Resinchem/ESP-Parking-Assistant/assets/157761983/f4088a71-57f7-4c7d-958d-46b01a027981)

![image](https://github.com/Resinchem/ESP-Parking-Assistant/assets/157761983/d9d31815-2a4e-4cf5-9d98-6253bc0df1fa)

![image](https://github.com/Resinchem/ESP-Parking-Assistant/assets/157761983/5ea829af-dabc-41f5-9eec-30a34e10a636)

I am glad to provide any information you need. Thanks.

**Reply by Resinchem** [reply]
OK... sorry for the basics.  Just wanted to eliminate anything simple at this point.  You may not want to take these next steps if the parking assistant is already installed and/or if you are not familiar with using the Arduino IDE, but you can try connecting the D1 Mini via USB and enabling serial monitoring.  You'd have to modify the .ino file and re-upload the modified sketch to do this.  And since the TF-Mini is also using UART, it would be disabled for this test, but if serial monitoring is enabled, you'd be able to see output of the D1 Mini connecting to wifi and MQTT.  It would output a message that it successfully connected to MQTT or that it failed and was unable to connect.  Then you'd have to disable serial monitoring again and re-upload to enable the TF-Mini again.  Yeah... it's a lot and you may not be willing to take it this far.

One other little thing just to try in the interim is to issue the OTA and reboot commands... just to verify that the D1 Mini is receiving (this is independent of MQTT).  You can enter in the IP address of your device in a browser followed by /otaupdate  or /restart (e.g. http://192.168.1.105/otaupdate) and see if the device responds.  For an /otaupdate, the LEDs should change to alternating red/green.  Don't worry if you don't upload anything.... after around 20 seconds without receiving a file, it will return to normal operation.  And of course a /restart will reboot the device and you'll see the normal LED start up sequence.  This is just to assure that the device is able to receive.

I'm at a bit of a loss at this point as to what could be occuring.  If you want to try the serial debug and need more info, just let me know.

**Reply by shomritab** [reply]
I am not familiar with Arduino IDE. Is there any tutorial or steps I can follow to do this. I installed the IDE but I am confused about how to proceed. 

The tried the other thing otaupdate/restart. they work just fine. 

There is one more issue though, which I am not sure is something to do with this. I am able to connect to the parkassist IP address only with my iPhone browser. But when I try the same with macbook browser, I doesn't connect, I tried a lot but could never connect.

**Comment by Resinchem** [comment]
If you aren't familiar with, and haven't used the Arduino IDE, this could be a little tough to get through.... as you'll need to install the ESP8266 board supports, the various libraries for the parking assistant, etc.  But if you want to take a shot, this article is a good starting point:  https://randomnerdtutorials.com/how-to-install-esp8266-board-arduino-ide/  Do note that you don't need an FTDI programmer in your case, because you can just use the USB port.

Out of curiosity, what IP address are you using to access the parking assistant web interface?  I'm wondering if there is some sort of network thing going on since you should be able to access the parking assistant from any device on the same network/subnet.  But I did note that your MQTT broker (and I'm assuming Home Assistant) are on a 192.168.4.x subnet.  Are you running VLANs or anything else unusual on your network?  And you are sure that the parking assistant is joining your wifi... and you aren't accessing via the initial broadcast hotspot.  Since the initial hotspot also uses 192.168.4.1, I'm wondering if the parking assistant isn't truly on your wifi network and therefore cannot talk to the broker.  It should be connected to your local wifi SSID and you should not be accessing it via the ESP_ParkingAsst hotspot.

**Reply by shomritab** [reply]
I have been onto this since this evening. And this finally solved it for me. I uploaded the sketch via arduino IDE and everything worked like magic. And then I tried to update it OTA with the .bin file and it went back to the same broken state. Not sure why this behavior, although both are basically the same firmware. Probably you can think of something :)  ( I am just a noob) 

To answer your other question - I was connecting to the IP 192.168.5.xxx to connect to the Parking assistant , but strangely with my macbook it wouldn't connect, this issue is also solved by the above workaround. 

** Although I had to rip off my installation to do all the testing. But it was worth it. Thanks for all the help.

**Reply by Resinchem** [reply]
That is a bit bizarre and I don't have an immediate answer.  The .ino and the .bin should be identical as the .bin is simply the compiled version.  The only thing that comes to mind is that when flashing the .bin, something wasn't working right with the data that is supposed to be written to SPIFFS... this is where the settings are stored.  But when you manually flashed the .ino file, that particular area did get formatted and written to correctly.  Just out of curiosity, what app did you use to initially flash the .bin file?  NodeMCU PyFlasher, something else?  I've never experienced what you have... and no one else has reported it, but if someone else does, I'd like to be able to share some possible reasons/fixes with them.

I apologize that you had to tear your install apart to get this fixed.  But I'm happy that you finally got it resolved.  Congrats!

**Reply by shomritab** [reply]
I used PyFlasher to do the initial flashing.

**Reply by Resinchem** [reply]
I just wanted to follow up with you on this as I just release an update (v0.45) that includes Home Assistant discovery, which will automatically add a device and related entities to Home Assistant without the need to create any YAML or do any configuration on the Home Assistant side.  As long as your MQTT broker information is correct and the parking assistant is able to communicate with your MQTT broker, all the Home Assistant integration will be done for you as soon as you enable MQTT discover via the parking assistant app.

Let me know your current status and if I can help out any further.

**Reply by shomritab** [reply]
Hello. Thanks for the update. I did an OTA update with v0.45 and enabled the MQTT discovery. It worked out fine. 
![image](https://github.com/Resinchem/ESP-Parking-Assistant/assets/157761983/edbb732d-8720-4bc6-a3c0-3d0f33e86023)

**Reply by Resinchem** [reply]
Outstanding... and thanks for letting me know.  Since I've only tested the MQTT discovery myself (and that's why I consider it beta at this point), I am looking for feedback from others to verify that the discovery process worked for them.  So it is really helpful that you let me know that it worked fine for you.

I'm working on some additional updates to the parking assistant that is going to upgrade the original wifi onboarding process and modernize how the configuration is stored... and maybe add an additional option or two, but not sure when that will be ready for release.  But I do really appreciate the feedback on the MQTT discovery process.

**Reply by shomritab** [reply]
That's the least I could do in return for all the knowledge I get from your videos :) Would be waiting for your next update.

**Reply by DeWaHa** [reply]
Hello,
I’ve installed 0.46 and my SSID problems are solved.
I was also able to integrate it in Home assistant via MQTT discovery.
Many thanks for all your efforts.

**Comment by jtroberts1** [comment]
I created a custom PCB a few years back with an integrated 74ahct125 and power module etc.., however I did not use D6 (PIN 12).  I could compile from your source, I just never had much luck with the arduino IDE and mostly stick to ESPHOME and Tasmota. I could probably fiddle with VsCode.

Ok my question, would it be as simple as changing line 35 from PIN 12 to PIN 13?  Anywhere else I would need to modify.

BTW: Great work.  I enjoy your videos

**Reply by Resinchem** [reply]
Thanks for the kind words!  Yes, you should be able to change the LED data pin in that one line.  The only other two pins that are used are the UART (GPIO1 and GPIO3).  So you can change it to any appropriate GPIO pin and recompile.  That variable is only used in one other location, and that is to setup the LEDs with FastLED.

Let me know if you have issues... but just change that line 35 to #define LED_DATA_PIN 13 then recompile and you should be good to go!

FYI:  I started out primarily with ESPHome and Tasmota as well, but picked up Arduino/C++ because I kept running into situations where I wanted to do more advanced things than could be done with the other two.

**Reply by jtroberts1** [reply]
Surprisingly Arduino IDE is much easier now when using .appimage on linux.  No issue with boards come etc... 

Any Idea what versions of the various libraries you used?

If you have your IDE still setup would you be able to compile with GPIO15 (D8) ?

My error if you see anything simple.  thx

`/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:336:65: warning: backslash and newline separated by space
  336 |   mainPage += "<label for=\"mm\">Millimeters</label>&nbsp;&nbsp;\
      |                                                                  
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:823:34: warning: extra tokens at end of #ifdef directive
  823 | #ifdef ARDUINOJSON_VERSION_MAJOR >= 6
      |                                  ^~
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:1005:220: warning: backslash and newline separated by space
 1005 |       discMsg += "If you do not see the new device under your Home Assistant MQTT integration, please use a utility (e.g. MQTT Explorer or similar) to see if the controller is successfully connecting to your broker.<br>\
      |                                                                                                                                                                                                                             
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:1019:144: warning: backslash and newline separated by space
 1019 |       discMsg += "Please use a utility (e.g. MQTT Explorer or similar) to test if the controller is successfully connecting to your broker.<br>\
      |                                                                                                                                                 
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:1033:144: warning: backslash and newline separated by space
 1033 |       discMsg += "Please use a utility (e.g. MQTT Explorer or similar) to test if the controller is successfully connecting to your broker.<br>\
      |                                                                                                                                                 
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:1047:141: warning: backslash and newline separated by space
 1047 |     discMsg += "Please use a utility (e.g. MQTT Explorer or similar) to see if the controller is successfully connecting to your broker.<br>\
      |                                                                                                                                              
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:1092:144: warning: backslash and newline separated by space
 1092 |       discMsg += "Please use a utility (e.g. MQTT Explorer or similar) to test if the controller is successfully connecting to your broker.<br>\
      |                                                                                                                                                 
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:1106:144: warning: backslash and newline separated by space
 1106 |       discMsg += "Please use a utility (e.g. MQTT Explorer or similar) to test if the controller is successfully connecting to your broker.<br>\
      |                                                                                                                                                 
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:1121:141: warning: backslash and newline separated by space
 1121 |     discMsg += "Please use a utility (e.g. MQTT Explorer or similar) to see if the controller is successfully connecting to your broker.<br>\
      |                                                                                                                                              
In file included from /home/jtroberts/Arduino/libraries/FastLED/src/FastLED.h:79,
                 from /media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:18:
/home/jtroberts/Arduino/libraries/FastLED/src/fastspi.h:157:23: note: '#pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output'
  157 | #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino: In function 'void handleRoot()':
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:470:15: warning: unknown escape sequence: '\;'
  470 |   mainPage += "<table>\
      |               ^~~~~~~~~
  471 |       <tr>\
      |       ~~~~~    
  472 |       <td><label for=\"mqttaddr1\">Broker IP Address:</label></td>\
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  473 |       <td><input type=\"number\" min=\"0\" max=\"255\" step=\"1\" name=\"mqttaddr1\" style=\"width: 50px\;\" value=\"";
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:475:15: warning: unknown escape sequence: '\;'
  475 |   mainPage += "\">.<input type=\"number\" min=\"0\" max=\"255\" step=\"1\" name=\"mqttaddr2\" style=\"width: 50px\;\" value=\"";
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:477:15: warning: unknown escape sequence: '\;'
  477 |   mainPage += "\">.<input type=\"number\" min=\"0\" max=\"255\" step=\"1\" name=\"mqttaddr3\" style=\"width: 50px\;\" value=\"";
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:479:15: warning: unknown escape sequence: '\;'
  479 |   mainPage += "\">.<input type=\"number\" min=\"0\" max=\"255\" step=\"1\" name=\"mqttaddr4\" style=\"width: 50px\;\" value=\"";
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:481:15: warning: unknown escape sequence: '\;'
  481 |   mainPage += "\"></td></tr>\
      |               ^~~~~~~~~~~~~~~
  482 |       <tr>\
      |       ~~~~~    
  483 |       <td><label for=\"mqttport\">MQTT Broker Port:</label></td>\
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  484 |       <td><input type=\"number\" min=\"0\" max=\"65535\" step=\"1\" name=\"mqttport\" style=\"width: 65px\;\" value=\"";
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:502:15: warning: unknown escape sequence: '\;'
  502 |   mainPage += "\"> (16 alphanumeric chars max - no spaces, no symbols)</td></tr>\
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  503 |       <tr>\
      |       ~~~~~    
  504 |       <td><label for=\"mqttperiod\">Telemetry Period:</label></td>\
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  505 |       <td><input type=\"number\" min=\"60\" max=\"600\" step=\"1\" name=\"mqttperiod\" style=\"width: 50px\;\" value=\"";
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino: In function 'bool reconnect_soft()':
/media/DATA/Downloads/IoT/Firmware/ParkingAssistant/ParkingAssistant.ino:1262:1: error: control reaches end of non-void function [-Werror=return-type]
 1262 | }
      | ^
cc1plus: some warnings being treated as errors

exit status 1

Compilation error: control reaches end of non-void function [-Werror=return-type]`

**Reply by Resinchem** [reply]
GPIO13 or GPIO15?  In the initial post, you mention that you want to use pin 13, but in this follow up, you say pin 15.  I can probably recompile the most recent version for you with a different GPIO pin, but need to know whether you want that pin to be 13 or 15.

**Reply by jtroberts1** [reply]
thx  GPIO15

**Reply by Resinchem** [reply]
OK... I compiled a version for GPIO15.  Unfortunately, I cannot attach .bin files here in the discussion (or issue) section.  What I have done temporarily is to add a binary to the latest release (0.46).  So if you go to the latest release from the main code page, you should see a new binary 'ParkingAsst_ESP8266_Pin15.bin'.  Download this file and flash it to the D1 Mini.  The LEDs will then use pin 15.

Please let me know when you have downloaded the file because I'd like to remove it from the release.  I won't be able to support multiple versions moving forward and simply won't have time to create custom binaries for everyone.

**Reply by jtroberts1** [reply]
You Rock sir!  worked perfectly.  Now I'll get things installed and tuned.  thx again

**Reply by Resinchem** [reply]
Fantastic!  Glad it worked and I'm happy to have helped a bit.  Thanks for letting me know.  I'm going to remove the _Pin15 binary from the release, but let me know if you need it again for some reason.

**Comment by pjoter1337666** [comment]
After the Arylic amp the next project in line was the parking assistant and I have to say it works better than I expected it to! I am now in testing phase and all seems to be in a good shape and the MQTT integration allowed me to set up an automation where after crossing a given threshold it plays a sound on the garage Arylic amp no less :) 
I have gone with the exposed electronics theme here as well, perhaps I'll try to hide the cables sticking out down the road but it is not a priority now. Also I have this round LED strip that looks nice as it fills up with yellow color when in the active zone. And it looks a bit like a traffic light when it's fully red. 

One note I would have for an update to consider is maybe to add an option to set a static IP at WiFi onboarding, since I am not allowed to enter my router's settings (thanks to my ISP) and I need to go into the browser and check all consecutive IPs after I reboot the router. 

So thank you very much again for your excellent work and two coffees are on the way! 

![IMG_20240725_085232](https://github.com/user-attachments/assets/73ae783c-3fef-4cf8-bf84-fced62955ecf)

**Reply by Resinchem** [reply]
Nice build!  Using the ring of LEDs is an interesting way to display the approach.  And I like the creative idea of using the Arylic amp for sound based on the approach.  It has been requested that I add sound via some sort of increasing buzzer/beeping sound, but I haven't really looked into that yet.  I don't think it would be that difficult to implement, I just haven't found the time yet.

As far as a static IP address, I think that is something that is also possible.  The existing WiFi library that I'm using needs to be replaced, as it has issues with some SSIDs that are either very long or contains special characters.  I have developed my own prototype for onboarding and connecting to WiFi, but it is still in early development and needs much more testing. I can certainly put it on the list to look at adding a static IP option for my own library.  Thanks for the suggestion!

**Reply by jtroberts1** [reply]
It would be nice to port this to ESP32.  Not sure if it's 8266 specific, but my sensors seem to have decent lag in the LED response.   Once it "wakes up" it works well.  I have to pull up real, real slow to get smooth feedback.  I don't think this is too complicated, we're just changing the board and pins.  I'll see if I can get it done but Arduino IDE hates me. :(

**Reply by Resinchem** [reply]
I've had other requests to port this to an ESP32, but it would require replacing a number of libraries and other changes and until I can find the time to integrate both the ESP32 and ESP8266 into a single build, I likely won't do so.  I don't want to have to maintain two different builds, one for the 8266 and one for the 32 moving forward as I potentially add features or roll out other updates.  Frankly this just isn't a high priority as the ESP8266 is perfectly capable of handling reading a sensor and lighting up the LEDs.  I really don't feel that the ESP32 would add anything... at least not until I do something like add a side lateral sensor to the system.

Regarding your "lag", I think it is likely due to the smaller number of LEDs.  The smaller number of LEDs that you have, the less "sensitivity" you will have.  For example, say you have a 12' approach zone and 12 LEDs.  Each LED will only light up for every foot of distance traveled.  This becomes more apparent as you approach the final park zone.  Depending on exact distances, the last "yellow" LED could be lit up for a full foot of travel before you hit the "red" park zone.  However, if you have say 36 LEDs, each LED is going to light up for every 4" of travel... giving you much better sensitivity, especially at the approach and parked zone interface.

One thing you might try if you haven't, is to use one of the full strip effects (as opposed to in-out or out-in).  When using full strip or full strip inverted, you are using the entire number of LEDs (in your case 24 I believe), as opposed to one of the in-out effects that splits that number in half, giving you only 12 LED sensitivity.  Try a full strip effect and it should come close to doubling the sensitivity you are seeing.  Let me know what you find, but I don't think the issue is due to either lag from the sensor or issues with an "underpowered" ESP8266.

**Reply by jtroberts1** [reply]
Your suggestions is exactly what I have done...
64 pixels (custom PCB no current concerns)
192 wake
190 active
41 parked distance
full strip effect

I increased the active zone to 190 when I switched to 64 pixels hoping to get a smoother effect.  NOTE: 64 individual pixels.  IE: Each pixel has an integrated chip versus some of the dense strips that are grouped.

**Reply by Resinchem** [reply]
Do you know the actual type of LEDs?  Are these 5V WS2812b or some other variant?  While the library I'm using supports multiple types of LED chips, the definition used in my code specifically uses WS2812b: 

```FastLED.addLeds<WS2812B, LED_DATA_PIN, GRB>(LEDs, NUM_LEDS_MAX);```

There may be unexpected behavior if using a different LED chip.  I'm certainly happy to help you try to troubleshoot.  I'm just trying to assure I fully understand your particular setup.  And as an FYI, increasing the active zone distance will actually _decrease_ the sensitivity.  Remember that the code is taking the total distance and dividing by the number of LEDs.  The longer the distance, the further that the object must travel before an additional LED lights up.  

I don't know if you have this installed in your garage yet or are still working on a benchtop, but if it is mounted, then try a simple experiment.   With your active distance currently set at 190, take some sort of flat surface (I always used something like a 2' x 2' piece of cardboard) and start only about 3-4' outside the parked zone distance and slowly approach the parked zone and note the sensitivity.  Now change that active zone to something less than 1/2 the original distance (say 80-90" or even a bit less) and repeat the same test from the same initial distance.  You should be seeing nearly double the sensitivity as you approach the parked zone... assuming something else isn't going on with your system.  Let me know what you find and we can take it from there.

**Reply by jtroberts1** [reply]
I absolutely appreciate it...   BTF WS2812B ECO Led Pixel Strip 100/m
This is the part # on the bag. WS2812b1m100lb30e

"And as an FYI, increasing the active zone distance will actually decrease the sensitivity."
but by adding a high count of LED's this should improve the sensitivity right?

**Reply by Resinchem** [reply]
OK... thanks for confirming that the LEDs are WS2812b.  That eliminates one possible source of problems.  And yes, you are correct... shortening the approach zone distance and/or adding more LEDs will increase the sensitivity... or both.  Doing either one of those means that a new LED will light up with a smaller change in distance.  And the full strip effects will have approximately double the sensitivity of the in-out effects, as it uses the full count of LEDs instead of just half of the count for the effects that split the strip in half.

**Comment by steveof2620** [comment]
Just a quick note to say thanks for the parking assistant device. Have it up and running in my garage and find it incredibly useful. Thanks also for the attention to the documentation that goes along with it. I ran into some problems which I was able to overcome thanks to the attention to the documentation you provided along with the system. Many thanks.

**Reply by Resinchem** [reply]
Thanks for passing along your success!  I love to hear it.  I still have a few ideas for the future for this project.

Was there anything in particular that you ran into that I could either address through documentation and/or through my troubleshooting section on the Wiki?  The most common issue I've heard about involves issues getting a good flash.   There were a lot of wifi-related issues early on before I replaced the external library with my own onboarding code.  Most of the other issues are related to wiring mistakes or failure to follow instructions.  Hence if there is a way to improve the documentation and help others avoid these common problems, I'd be happy to add it!

Thanks again for letting me know about another successful install.  Out of curiosity, are you using the ESP8266 or ESP32 version?  And if using ESP32, did you add the side/lateral sensor?  Mine has been working fine, but both ESP32 support and side sensing are pretty new so I don't have a of 'real world' results from others yet.

**Reply by steveof2620** [reply]
No all good. I tried to flash an OTA upgrade to my existing system the other day and screwed it up cause I didn't read your instructions on version compatibility properly. Couldn't get a flash going to install the new version on my demos d1 mini (8266) until I read your post in trouble shooting re the TF-Mini. Once I disconnected it, it was good as gold. Thanks again.
